File: lista 1 cwiczenia ocaml/zadanie2.ml
let rec count ( x , xs) =
	if xs=[] then 0
	else if List.hd xs = x then 1 + count(x , List.tl xs)
	else count (x,List.tl xs);;

count ('a',['a';'b';'c';'a';'a';'c';'d';'a'])
	
	 

File: lista 1 cwiczenia scala/src/zadanie2.scala


class zadanie2 {
  
  
  def count [A] ( x:A , xs:List[A] ):Int =
    if (xs.isEmpty) 0
    else if ( xs.head == x ) 1 + count( x, xs.tail )
    else count( x, xs.tail )
  
  
  count ( 'a' , 'a'::'b'::'a'::Nil )
  
  
  
  
  
  
}